### ISSUE TITLE
New issue/behaviour

### ISSUE THREAD

**Issue by CristianGrecu** [fgl27/SmartTwitchTV#237]
Hi, so my app starting doing something that I haven't seen before. As usual it's either me, Twitch or the app.

When watching some streams the screen would go black for a second with the buffering wheel showing and then back to normal. Sometimes it happens a a lot of times without any pattern. The speed is good, no dropped frames, nothing showing in the top right stats. Most of the times it comes back, but randomly I can see error code 2004. No one else seems to say anything in chat. Usually this happens when the stream ends for everyone if there's a serious problem. 

Since it only happens with certain streams and no pattern (sometimes every few minutes, could even happen after half an hour) I assume this is Twitch again, right? The other issue that they admitted to still hasn't been fixed.

I wasn't sure if I should ask you, but I haven't seen this before and I've been using the app since 2022.

Thank you!

**Comment by fgl27** [comment]
Yes, Twitch random server issue...

2004 is a network-related error:

![image](https://github.com/user-attachments/assets/fea98775-eb3c-4c67-9fde-f1f84df42929)

as the error informs, the server returns something unexpected, even if all is fine there may be a random connection issue with the server providing the stream

This is also maybe related to how they randomly limit the speed of the connection.

**Comment by CristianGrecu** [comment]
That's just great, I've been using a VPN to fix the broken streams (many people have the same issue here in Eastern EU when watching US streams). Now the good streams start falling apart... 

Thank you for the fast answer.

**Comment by s1p2a3m** [comment]
this is becoming super annoying, please reconsider adding a retry or something, because it throws this error and you have to click the stream again and again and again... it does not happen on every stream

**Comment by CristianGrecu** [comment]
When it was happening to me it usually goes from black screen back to the stream on its own, but you are right, sometimes I have to press ok a bunch of times to stay.

I think they are doing some infrastructure changes because my latency to Twitch went from 50 to 10 on all streams and that's when the issues started. They also admitted they know about the problems. Not much we can do.

**Comment by CristianGrecu** [comment]
No no, sorry I didn't explain correctly. There are 2 issues right now. The one I told you about a while ago, the streams that have low net speed in the top right corner. That's an issue a lot of people have when watching from EU to US servers.

I only use a VPN to route the connection through Seattle for those US streams. Most EU streams used to work perfectly, but some developed this black screen buffer. No VPN was active when it happened.

As I said in the previous message, my ping to twitch dropped from 45-50 to 10-15. Does that mean that they are installing new servers closer to me or? Maybe it had nothing to do with it.

So to summarise my 200 ping streams through VPN work fine, the black screen issue happened on an EU stream, no VPN.

**Comment by CristianGrecu** [comment]
Oh and it's not a temporary thing or related to a streamer's internet. It's been doing it today too, on a completely different stream that I haven't watched in weeks. It's random as usual...

**Comment by lextra2** [comment]
US streams are basically unwatchable for me right now. "Buffering" every 5 minutes followed by the stream cancelling itself...

**Comment by CristianGrecu** [comment]
I found a few that do this and tested a bit. VPN actually does seem to fix them, they run much better.

I use Nord VPN and route through Seattle, but please don't buy an expensive one just because I said it works.

So far I've had to pay for 3 months of VPN because Twitch never fixed my original issue...

**Comment by CristianGrecu** [comment]
Also, I was watching a stream just now and someone said : "is it me or the stream stopped", so there's definitely something going on.

**Comment by fgl27** [comment]
From Brazil is all fine, US and EU streams have no issues...

The ping is not direct to the Twitch server serving the stream but to Twitch itself, as their servers don't allow ping

But works as a metric to know if there is a network delay related to Twitch.

The Twitch site has an auto-recovery process, I watch from the site on a computer in a VPN and I have issues there, buffers and random freezes, but I see the player just recovering by resetting the stream or just retrying and retrying.

**Comment by fgl27** [comment]
The way the player is configured today it already retries, it will try 3 times if fails due to an error, during this tries you probably see a back screen with a buffer spinner, and after that, it shows the error **Player is lagging, quality changed to 'Auto mode' Error Code: XXXX**

At this moment it tries to reset the stream and enable the AUTO playback...

If it fails here is because the connection was unuseful

this can be because Twitch denied the connection, or there is no way to connect to the server...

Then after a few seconds, all is fine you can open the stream again, and works.

This is what you all are getting right?

I can try to improve this but as the app is already doing a lot may not be possible.

Let me know if the process described there is the one you have so I can think on a solution.

**Comment by CristianGrecu** [comment]
Yes, exactly what you described, but on top of that, most of the times if a stream manages to "recover" on its own it also comes back with a 7+ second delay (at least for me) . So I have to adjust it with the remote anyway.

Today some streams worked perfectly, some almost unwatchable, at roughly the same time. I am sure it's not my internet. Also the fact that VPN actually fixes it.

**Comment by lextra2** [comment]
> The way the player is configured today it already retries, it will try 3 times if fails due to an error, during this tries you probably see a back screen with a buffer spinner, and after that, it shows the error **Player is lagging, quality changed to 'Auto mode' Error Code: XXXX**
> 
> At this moment it tries to reset the stream and enable the AUTO playback...
> 
> If it fails here is because the connection was unuseful
> 
> this can be because Twitch denied the connection, or there is no way to connect to the server...
> 
> Then after a few seconds, all is fine you can open the stream again, and works.
> 
> This is what you all are getting right?
> 
> I can try to improve this but as the app is already doing a lot may not be possible.
> 
> Let me know if the process described here is the one you have so I can think of a solution.

Yes but without the error code. (I have auto-switch resolution on lag disabled) It just closes the stream as if the streamer went offline.

**Comment by fgl27** [comment]
OK I'll see what I can do to improve those scenarios retry and latency

Thanks

**Comment by CristianGrecu** [comment]
Overall It's been a bit better for me today, only happened on one stream, but the worst example so far, kept going in a black screen loop even after I prssed "stay on stream".

**Comment by fgl27** [comment]
> Overall It's been a bit better for me today, only happened on one stream, but the worst example so far, kept going in a black screen loop even after I prssed "stay on stream".

you don't see the 'Checking if is live ...' ever in that scenario?

**Comment by fgl27** [comment]
You can enable this option to auto always stay on the stream

![Screenshot_2024-11-23-20-59-07-19_8fb930a926872f0234df725c011c783b](https://github.com/user-attachments/assets/7c970c7f-57f1-4a2a-a63a-086dbb68b5c5)

That is the only option for this type of problem

Is it fails again and again is because you have no path to the server

Exit, refresh and opening it again doesn't fix that, is just time passing and now whenever was your internet issues is now gone 

The only information the app uses to get the stream is the streamers name something that doesn't change after any refresh

Not much to do here other then try yours ISP maybe they can do something.

**Comment by fgl27** [comment]
OK, for now, who has this issue can you try to enable these setting options?

**Settings > Player end dialog options > Always stay with... set to YES**

![image](https://github.com/user-attachments/assets/4df41953-756b-4eb4-9aac-e4fd483a5c7e)

![image](https://github.com/user-attachments/assets/bcc866f5-83d4-4575-ba9d-69b42f0ef4cc)

If I make any change it will be the same as enable that...

In resume...

Error, the app will try to recover 3 times after that will try to change the quality to auto if allowed, else will just end the stream

If always stay is enabled it will... check if the stream info shows the streamer is live if yes will retrieve a new stream playback info and try to playback

**Comment by CristianGrecu** [comment]
I would try it, but I couldn't find any bad streams today. Maybe the others that posted here still have problems.

**Comment by CristianGrecu** [comment]
I was also curious, does your app have any way to pull info like the Chrome console does? Like when Twitch support ask you for a specific video/server id to troubleshoot.

**Comment by fgl27** [comment]
> I was also curious, does your app have any way to pull info like the Chrome console does? Like when Twitch support ask you for a specific video/server id to troubleshoot.

The server is geo base so is the same as in the browser.

The app can see the value but isn't exported to the user.

**Comment by lextra2** [comment]
Its much better today. Only happened to one stream in 5 hours.
Will probably fix itself within one week.

**Comment by CristianGrecu** [comment]
I don't know about everyone else, but I haven't had any issues for quite a few days. Even my other problem is mostly gone, didn't need the VPN at all.

You can close this if you want, never had anything to do with the app anyway.

Let's hope I didn't jinx it lol.
